tests/integration/lens.go

```diff
@@ -75,9 +75,34 @@ func getMigrations(
 
 		configs, err := db.LensRegistry().Config(s.ctx)
 		require.NoError(s.t, err)
-
-		// The order of the results is not deterministic, so do not assert on the element
-		// locations.
-		assert.ElementsMatch(s.t, configs, action.ExpectedResults)
+		require.Equal(s.t, len(configs), len(action.ExpectedResults))
```

Comment:
This is due to the http client losing type info when serializing to json. The newly added `assertResultsEqual` function must be used to properly compare the two values.

Reply:
You mean the http stuff is returning values of a type other than the one expected in the test(s)?

Reply:
The underlying values are the same but the types differ slightly. 

A good example is when you serialize a number to json. It loses its type value of uint64 or float64 and becomes a string representation. When it's deserialized it is stored as an intermediary type `json.Number`. 

This allows the test suite to check the expected type and attempt to convert the `json.Number` to the correct type before comparing.

Reply:
Asserting the type is pretty important, particularly for the embedded Go client.  Changing the return type is a breaking change.

It also at glance looks like you are equating zero and option.None too?  This is also incorrect, including within http/gql contexts (think zero attendees vs undefined/nil - zero represents the knowledge that no one is attending, nil means that it is unknown).

Reply:
I've updated the `assertResultsEqual` to use the new "relaxed" comparison only when the client type is not `goClientType`.

https://github.com/sourcenetwork/defradb/blob/5f96d78080d65af8b7b24e787e19f62c1b9b21c5/tests/integration/results.go#L26-L38

Reply:
This still makes me uncomfortable:

1) The type assertion is opt-in, not opt-out, it would be very easy to accidentally not test the typing of new client interface implementations.  It would prefer that the `default` is `assert.Contains`.

2) The return types of all client interface implementations is important, and should be deterministic and consistent.  The Go client is just the most likely to be used directly by users.  That said, `http`, and later, `cli`, are new implementations, and it might be sensible to add a new ticket to the backlog to revisit this later, as to not block the main body of work in this PR.

Reply:
I've updated the default comparisons like you recommended.

I'm still thinking of a solution for your second point, but it will most likely be addressed in a follow up PR.

Reply:
Cheers Keenan, and really no pressure to sort out (2) within this PR.